packages/ember-glimmer/tests/integration/content-test.js

```diff
@@ -584,4 +585,236 @@ moduleFor('Dynamic content tests (integration)', class extends RenderingTest {
 
     this.assertText('');
   }
+
+  ['@test quoteless class attributes update correctly']() {
+    this.render('<div class={{if fooBar "foo-bar"}}>hello</div>', {
+      fooBar: true
+    });
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('foo-bar') } });
+
+    this.runTask(() => this.rerender());
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('foo-bar') } });
+
+    this.runTask(() => set(this.context, 'fooBar', false));
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello' });
+
+    this.runTask(() => set(this.context, 'fooBar', true));
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('foo-bar') } });
+  }
+
+  ['@test quoted class attributes update correctly'](assert) {
+    this.render('<div class="{{if fooBar "foo-bar"}}">hello</div>', {
+      fooBar: true
+    });
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('foo-bar') } });
+
+    this.runTask(() => this.rerender());
+
+    this.assertElement(this.firstChild, { tagName: 'div', content: 'hello', attrs: { 'class': classes('foo-bar') } });
+
+    this.runTask(() => set(this.context, 'fooBar', false));
+
+    // HTMLBars differs in behavior here as it leaves the empty
+    // class attribute.
+    assert.equal(this.firstChild.className, '');
```

Comment:
(I don't think we need to block this PR for that, but we should probably go back to glimmer and fix it)